### PR TITLE
M2-5380: Conditional logic improvement

### PR DESCRIPTION
**Pull request by vmkhitaryanscn**
### 📝 Description

🔗 [Jira Ticket M2-5380](https://mindlogger.atlassian.net/browse/M2-5380)

This PR adds optimizations for conditional logic
Calling `const currentStorageRecord = getCurrentActivityStorageRecord()!` on every iteration on bigger pipelines was causing lags.

This PR changes:
-  Conditional logic improvement

### CODE REVIEW

**Review by farmerpaul — APPROVED** [review]
I compared performance of progressing through questions of the Healthy Brain Network activities between `dev` and your PR, and it's definitely much faster in your PR – nice work! Code changes look sensible to me so I'll approve, however I think it would be wise for someone more familiar with the code to approve as well.

**Review by BamMironov — APPROVED** [review]
Looks much better than the previous implementation. Thanks

File: src/features/pass-survey/model/hooks/useActivityState.ts [hunk 1]
```diff
@@ -171,43 +171,31 @@ function useActivityState({
     });
   }
 
-  function removeAnswer(step: number) {
-    const currentStorageRecord = getCurrentActivityStorageRecord();
-
-    if (!currentStorageRecord) {
-      return;
-    }
+  function removeConditionallyHiddenItemsAnswersAndTimers(fromStep: number) {
```
Comment (anq83) [inline src/features/pass-survey/model/hooks/useActivityState.ts:174]: I like the idea to combile callback and iteratePipeline, but 
1. let's keep the previous name - iteratePipeline or iterateWithConditionalLogic
2. it seems this snippet is not considered:
```
if (!currentStorageRecord?.timers) {
   return;
}
```

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/features/pass-survey/model/hooks/useActivityState.ts (hunk 1 above)
Comment (anq83) [reply src/features/pass-survey/model/hooks/useActivityState.ts]: 2. yes, but not the same. In the prev. version it was just return, so {} wasn't recorder into the object unlike the current version. 
For example: !!undefined - false, !!{} - true.

**Review by anq83 — APPROVED** [review]

**Pull request merged**